### ISSUE TITLE
[Feature] Add a skip request function

### ISSUE THREAD

**Issue by davidsebag0705** [usebruno/bruno#2116]
### I have checked the following:

- [X] I've searched existing issues and found nothing related to my issue.

### Describe the feature you want to add

It will be useful to skip a request programmatically for example by testing a variable set by a previous request.  
It could be useful to build more complex testing scenario.

### Mockups or Images of the feature

This feature exists now in [Postman](https://community.postman.com/t/the-new-skip-request-feature/54806)

**Comment by busy-panda** [comment]
Yes, it is useful indeed, we also use it in postman.

So it could be the following syntax to be close to postman syntax:

```javascript
// set a local variable to true
// change to false to see all tests get executed
const skipTest = true;

// this test will always be executed
test("Status code is 200", function () {
    expect(res.getStatus()).to.equal(200);
});

// this test will be skipped when skipTest is set to true
(skipTest ? test.skip : test)('Response time is less than 200ms', function () {
    expect(res.getResponseTime()).to.be.below(200);
});
```

The skipped tests need to be counted in reports like the failed and passed tests.

**Comment by JorgeTrovisco** [comment]
Hi i studied a solution and need some inputs

# 1 - Update Bru on [packages/bruno-js/src/bru.js](https://github.com/usebruno/bruno/blob/77b1e6d738187f6b5bec6e00d66a5f4728152832/packages/bruno-js/src/bru.js)

Next to
````js
  setNextRequest(nextRequest) {
    this.nextRequest = nextRequest;
  }
````

Add something like
````js
  skipRequest() {
    this.skipRequest = true;
  }
````

# 2 - Update ScriptRuntime on [packages/bruno-js/src/runtime/script-runtime.js](https://github.com/usebruno/bruno/blob/77b1e6d738187f6b5bec6e00d66a5f4728152832/packages/bruno-js/src/runtime/script-runtime.js)

Exclusively on the function ``runRequestScript`` update the return to 

````js
    return {
      request,
      envVariables: cleanJson(envVariables),
      collectionVariables: cleanJson(collectionVariables),
      nextRequestName: bru.nextRequest,
      skipRequest : bru.skipRequest
    };
````


# 3 - Update the renderer:run-collection-folder on [packages/bruno-electron/src/ipc/network/index.js](https://github.com/usebruno/bruno/blob/77b1e6d738187f6b5bec6e00d66a5f4728152832/packages/bruno-electron/src/ipc/network/index.js#L812)


Before

````js
    if (preRequestScriptResult?.nextRequestName !== undefined) {
        nextRequestName = preRequestScriptResult.nextRequestName;
    }
````

Add 

````js
    if (preRequestScriptResult?.skipRequest) {
        continue;
    }
````

---

## Bonus:

Would be great to add the translation on [packages/bruno-app/src/utils/importers/translators/postman_translation.js](https://github.com/usebruno/bruno/blob/77b1e6d738187f6b5bec6e00d66a5f4728152832/packages/bruno-app/src/utils/importers/translators/postman_translation.js)

From
``pm.execution.skipRequest();``


To
``bru.skipRequest();``

**Comment by JorgeTrovisco** [comment]
Hi, i made the improvement, how can i request publish access to create the feature branch and create the PR?

Check the result bellow

![image](https://github.com/usebruno/bruno/assets/23103682/34a91715-4436-4b6f-81d7-716bcb287a65)
![image](https://github.com/usebruno/bruno/assets/23103682/f9e483b3-60bb-4ef7-9edb-1475e8b9b409)
![image](https://github.com/usebruno/bruno/assets/23103682/bed727b2-aa78-4f0d-9ba4-f730bec00ae5)

**Comment by JorgeTrovisco** [comment]
@helloanoop @lohxt1 I would like to contribute with this feature, how can i request push permissions?

What i got:

![image](https://github.com/usebruno/bruno/assets/23103682/00b2f7b8-09fe-46f1-a290-c7cc484298be)
![image](https://github.com/usebruno/bruno/assets/23103682/bde90487-0a02-4882-9709-642feaddcd98)
![image](https://github.com/usebruno/bruno/assets/23103682/25cf8a9d-ef3b-45fe-a736-364ffbd39bba)

**Comment by ben-sai** [comment]
Looks like this has been built some months ago and awaiting to be merged into the main release. do we need the community to raise this visibility so that it can get prioritized?

**Comment by JorgeTrovisco** [comment]
Hi @ben-sai,

As you can see, @martinsefcik has suggested an improvement in the PR. Unfortunately, I haven’t been able to implement it in the past few weeks due to time constraints. However, I believe it would be best to incorporate this improvement before merging, so the feature can be at its best for all of us.

**Comment by SPIRIT-00** [comment]
Confirm, important function.

**Comment by Its-treason** [comment]
> What's the status on this issue? I would like to have a skip functionallity

It's implemented as `bru.runner.skipRequest()`.

**Comment by hartmark** [comment]
> > What's the status on this issue? I would like to have a skip functionallity
> 
> It's implemented as `bru.runner.skipRequest()`.

Cool, I guess I need to wait for this to get merged before I can use it without getting the source?

**Comment by Its-treason** [comment]
> > > What's the status on this issue? I would like to have a skip functionallity
> > 
> > 
> > It's implemented as `bru.runner.skipRequest()`.
> 
> Cool, I guess I need to wait for this to get merged before I can use it without getting the source?

No, it's already released in version 1.38.0. It was merged in https://github.com/usebruno/bruno/pull/3719

**Comment by frille7070** [comment]
Can we have the skip function also for CLI please. It would be very useful

**Comment by sreelakshmi-bruno** [comment]
Hi, we have a fix in place for the CLI. This will likely go out in the upcoming release. Will post updates here.

**Comment by frille7070** [comment]
> Hi, we have a fix in place for the CLI. This will likely go out in the upcoming release. Will post updates here.

Very nice, thank you
